packages/router-generator/src/utils.ts

```diff
@@ -361,86 +518,54 @@ export function removeLastSegmentFromPath(routePath: string = '/'): string {
   return segments.join('/')
 }
 
+const nonNestedSegmentRegex = /_(?=\/|$)/g
+const openBracketRegex = /\[/g
+const closeBracketRegex = /\]/g
+
+/**
+ * Extracts non-nested segments from a route path.
+ * Used for determining parent routes in non-nested route scenarios.
+ */
+export function getNonNestedSegments(routePath: string): Array<string> {
+  nonNestedSegmentRegex.lastIndex = 0
+  const matches = [...routePath.matchAll(nonNestedSegmentRegex)]
+  const result: Array<string> = []
+  for (const match of matches) {
```

Comment:
I'm not sure you need to copy this into an array, the result of `.matchAll` should be iterable